File: factor-zero/golang/zero.go
package main

import "fmt"

func trailingZeroes2(n int) int {
	if n < 5 {
		return 0
	}
	return n/5 + trailingZeroes2(n/5)
}

func countOf5(n int) int {
	var result int
	for n%5 == 0 {
		result++
		n = n / 5
	}
	return result
}

func trailingZeroes(n int) int {
	if n < 5 {
		return 0
	}
	var count5 int
	for i := 5; i <= n; i = i + 5 {
		if i%25 == 0 {
			count5 += countOf5(i)
		} else {
			count5++
		}
	}

	return count5
}

func main() {
	var n int

	fmt.Scanf("%d", &n)
	// fmt.Println(trailingZeroes(n))
	fmt.Println(trailingZeroes2(n))
}

File: factor-zero/zero.c
#include "stdio.h"

int countOf5(int n ) {
	int result = 0;
	while ( n % 5 == 0 ){
		result++;
		n = n / 5;
	}
    printf("%d", result);
	return result;
}

int trailingZeroes(int n) {
	if (n < 5) {
		return 0;
	}
	int count5 = 0;
	for(int i = 5; i <= n; i += 5){
        if (n % 25 == 0)
        {
    		count5 += countOf5(i);
        }
        count5++;    
	}

	return count5;
}

int main() {
	int n;
	scanf("%d", &n);

	printf("%d", trailingZeroes(n));
}
